extension/lib/magento/Request.js

```diff
@@ -0,0 +1,115 @@
+const MagentoEndpointNotFoundError = require('./errors/EndpointNotFound')
+const MagentoEndpointNotAllowedError = require('./errors/EndpointNotAllowed')
+const MagentoEndpointError = require('./errors/Endpoint')
+const UnauthorizedError = require('../shopgate/errors/UnauthorizedError')
+const UnknownError = require('../shopgate/errors/UnknownError')
+const util = require('util')
+const _ = {
+  get: require('lodash/get')
+}
+
+/**
+ * All needed methods to fire requests to magento
+ */
+class Request {
+  /**
+   * @param {StepContext} context
+   * @param {string} token
+   */
+  constructor (context, token) {
+    this.logger = context.log
+    this.request = context.tracedRequest('magento-favorites-extension:request', { log: true }).defaults({
```

Comment:
Did you test this on sandbox stage, also if the tracing is still working? It might work but I am a bit scared since I know the tracedRequest method internally also uses .defaults() in an undocumented way and I am not sure about the side effects taking place here. We should thoroughly check that this is fine.

Reply:
Yes, this is used extensively in the getToken call in the magento-user extension: 
https://stash.localdev.cc/projects/SGX/repos/magento-user/browse/extension/helpers/tokenHandler.js